omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_4/FrontendJsonConfigController2_4.java

```diff
@@ -0,0 +1,155 @@
+import org.codehaus.jackson.map.ObjectMapper;
+import org.openmrs.module.ModuleException;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/frontend/config.json")
+public class FrontendJsonConfigController2_4 extends BaseRestController {
+
+    public static final String DEFAULT_FRONTEND_DIRECTORY = "frontend";
+    public static final String GP_LOCAL_DIRECTORY = "spa.local.directory";
+    private static final String JSON_CONFIG_FILE_NAME = "config.json";
+
+    private Log log = LogFactory.getLog(this.getClass());
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void saveFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String basicAuth = request.getHeader("Authorization");
+        boolean isAuthenticated = Context.isAuthenticated();
+        if (!isAuthenticated && basicAuth == null) {
+            response.sendError(HttpServletResponse.SC_FORBIDDEN, "No Authorisation Provided");
+            return;
+        }
+        if (!isAuthenticated && isValidAuthFormat(response, basicAuth)) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid credentials provided");
+            return;
+        }
+        User user = Context.getAuthenticatedUser();
+        if (user == null || !user.isSuperUser()) {
+            log.error("Authorization error while creating a config.json file");
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        }
+        saveJsonConfigFile(request, response);
+    }
+
+    private void saveJsonConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        try {
+            BufferedReader reader = request.getReader();
+            StringBuilder stringBuilder = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                stringBuilder.append(line);
+            }
+            String requestBody = stringBuilder.toString();
+
+            new ObjectMapper().readTree(requestBody); // verify that is in a valid JSON format
+
+            InputStream inputStream = new ByteArrayInputStream(requestBody.getBytes(StandardCharsets.UTF_8));
+            OutputStream outStream = Files.newOutputStream(jsonConfigFile.toPath());
+            OpenmrsUtil.copyFile(inputStream, outStream);
+
+            if (jsonConfigFile.exists()) {
+                log.debug("file: '{" + jsonConfigFile.getAbsolutePath() + "}' written successfully");
+                response.setStatus(HttpServletResponse.SC_OK);
+            }
+        } catch (JsonProcessingException e) {
+            log.error("Invalid JSON format", e);
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    private boolean isValidAuthFormat(HttpServletResponse response, String basicAuth) {
```

Comment:
Why do we even have an authorization check here? This can be handled by the filter in this module and we should be able to just reject things if there's no user or the user isn't an administrator.

Reply:
I have some doubts regarding our current authorisation filters.  
When i execute a request with/without an Auth header, it fails the first time but succeeds on subsequent tries, hence i resorted to retaining the authorisation confirmation logic.  However, i am going to remove it.